javascript/ql/src/Statements/UselessConditional.ql

```diff
@@ -17,18 +17,27 @@ import semmle.javascript.RestrictedLocations
 import semmle.javascript.dataflow.Refinements
 import semmle.javascript.DefensiveProgramming
 
+/**
+ * Gets the unique definition of `v`.
+ *
+ * If `v` has no definitions or more than one, or if its single definition
+ * is a destructuring assignment, this predicate is undefined.
+ */
+VarDef getSingleDef(Variable v) {
+  strictcount(VarDef vd | vd.getAVariable() = v) = 1 and
+  result.getTarget() = v.getAReference()
```

Comment:
I think this line is what's causing the new TPs.

`result.getTarget() = v.getAReference()` rejects destructuring patterns, whereas `isSymbolicConstant` below would previously admit those.

Reply:
Oh, indeed, very well observed. @esben-semmle, what do you think about this change?

Reply:
Well, it could be a `const` where the initializer is not something we recognise as a constant. In that case we still don't want to flag it, I think.

Reply:
I have created an issue for this discussion: https://jira.semmle.com/browse/ODASA-7637.
Happy to merge as is.